src/System.Net.Http/tests/FunctionalTests/ManagedHandlerTest.cs

```diff
@@ -391,5 +391,96 @@ public sealed class ManagedHandler_HttpClientHandler_ConnectionPooling_Test : Ht
                 }
             }
         }
+
+        [OuterLoop]
+        [Fact]
+        public async Task HttpConnectMethod_Succeeds()
+        {
+            using (HttpClient client = CreateHttpClient())
+            using (var listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
```

Comment:
There's a LoopbackServer test class that simplifies some of this, so you don't have to listen yourself explicitly etc.  Might be easier to use that.